### PR TITLE
Fix overlay start

### LINKED ISSUE #3515: Game overlay should be disabled by default
You need to restart your game after disabling the game overlay because it is injected, and that is quite frustrating to each user that doesn't appreciate the game overlay. Thus it should be disabled by default.

### LINKED ISSUE #3552: Mumble asks for permissions to access every application that opens (not working)
When mumble is open and other applications are in use, OSX sends a warning that mumble is trying to access the applications data and prompts to allow access. These are two examples of the prompts:

screen shot 2018-11-26 at 8 39 37 am
screen shot 2018-11-26 at 8 39 37 am

### PR DESCRIPTION
**Pull request by Krzmbrzl**
See commit messages for what is being done.

Fixes #3552 
Fixes #3515

### CODE REVIEW

File: src/mumble/Overlay.cpp [hunk 1]
```diff
@@ -185,11 +185,45 @@ QRectF OverlayGroup::boundingRect() const {
...
+
+Overlay::~Overlay() {
+	setActive(false);
+	if (d) {
+		delete d;
+	}
+
+	// Need to be deleted first, since destructor references lingering QLocalSockets
+	foreach(OverlayClient *oc, qlClients)
+	{
```
Comment (davidebeatrici) [inline src/mumble/Overlay.cpp:205]:
```suggestion
	foreach(OverlayClient *oc, qlClients) {
```

**Comment by migaspar** [comment]
the latest build is still not running :-(

```
Dyld Error Message:
  Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
  Referenced from: /Applications/Mumble.app/Contents/MacOS/Mumble
  Reason: image not found

Binary Images:
       0x10f616000 -        0x10fee0fff +net.sourceforge.mumble.Mumble (1.4.0~370~g165fe09~snapshot) <863334BF-43E3-342B-A24E-CBBB0BD5E030> /Applications/Mumble.app/Contents/MacOS/Mumble
       0x117d37000 -        0x117dc7cb7  dyld (733.8) <EBC07CB6-870A-3A8E-B48A-67F62EA161F3> /usr/lib/dyld
    0x7fff375fc000 -     0x7fff375fcfff  com.apple.ApplicationServices (48 - 50) <C1A1ED89-4F27-30E4-B53B-34062E971294> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices
    0x7fff3792e000 -     0x7fff37a0effa  com.apple.audio.toolbox.AudioToolbox (1.14 - 1.14) <7B308BA7-F08A-3E70-A494-F03A688AAA42> /System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox
    0x7fff37a10000 -     0x7fff37a10fff  com.apple.audio.units.AudioUnit (1.14 - 1.14) <C3937FFF-1BCA-3A4F-B718-7D6CB4903907> /System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit
    0x7fff3818d000 -     0x7fff3818dfff  com.apple.Carbon (160 - 162) <916D6D9E-3FE9-355C-913B-B44CAF35146D> /System/Library/Frameworks/Carbon.framework/Versions/A/Carbon
    0x7fff38672000 -     0x7fff38672fff  com.apple.Cocoa (6.11 - 23) <DDDE2ECF-6FD6-3EC7-B2F9-C1873F85D0E5> /System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa
    0x7fff38b5b000 -     0x7fff3906affa  com.apple.audio.CoreAudio (5.0 - 5.0) <C1EE25DD-6880-3FFB-8C7D-4CBC10B41BD2> /System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio
    0x7fff45c27000 -     0x7fff45c3fffd  com.apple.ScriptingBridge (1.4 - 79) <01931888-0387-34F7-AD76-BCCBC63BBA5B> /System/Library/Frameworks/ScriptingBridge.framework/Versions/A/ScriptingBridge
    0x7fff45c40000 -     0x7fff45f92ffa  com.apple.security (7.0 - 59306.80.4) <FA5EBA16-2B2B-3105-8A17-C8624C4B3BB3> /System/Library/Frameworks/Security.framework/Versions/A/Security
    0x7fff4601d000 -     0x7fff4604afff  com.apple.securityinterface (10.0 - 55139.0.1) <D2572186-FC98-30DE-B5F6-DA4B6CB5FA16> /System/Library/Frameworks/SecurityInterface.framework/Versions/A/SecurityInterface
    0x7fff6fdb2000 -     0x7fff6fdbffff  libxar.1.dylib (420) <317EA455-78EE-35A4-B63F-D85993B8697C> /usr/lib/libxar.1.dylib
```
let me know, if you need more information.

**Comment by Krzmbrzl** [comment]
Yeah I know. The installer that is currently build on the mac CI is unusable for non-dev systems. This change will be in the 1.3.1 release candidate so you'll be able to test the feature with that :)

Thanks for checking up on this though :+1:

File: src/mumble/Overlay.cpp [hunk 2]
```diff
@@ -185,11 +185,45 @@ QRectF OverlayGroup::boundingRect() const {
...
+		platformInit();
+		forceSettings();
+	
+		m_initialized.store(true);
+	}
+
+	setActiveInternal(act);
+}
+
+void Overlay::createPipe() {
```
Comment (Kissaki) [inline src/mumble/Overlay.cpp:225]: *Regression:* `createPipe()` is introduced but never called.

Comment (Krzmbrzl) [reply src/mumble/Overlay.cpp:225]: Good catch - (hopefully) fixed in #4281

**Pull request merged**